### PR TITLE
Added Azure.ContainerApp.APIVersion

### LINKED ISSUE #2398: Azure Container Apps control plane API versions 2022-06-01-preview and 2022-11-01-preview will be retired
### Existing rule

_No response_

### Suggested rule

The two spesific API versions `2022-06-01-preview` and `2022-11-01-preview` for Azure Container Apps will be retired on the November 16, 2023.

Management operations using the retired API versions may fail after November 16, 2023.

The rule should check for these versions and suggest migrating to a supported version if there is a match.

### Pillar

Operational Excellence

### Additional context

https://azure.microsoft.com/en-us/updates/retirement-azure-container-apps-preview-api-versions-20220601preview-and-20221101preview/

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2398

Added Azure.ContainerApp.MigrateAPIVersion rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
Hi @BernieWhite. Let me know what you're thinking. Rule number 400 btw 🥳

File: tests/PSRule.Rules.Azure.Tests/Resources.ContainerApp.json [hunk 1]
```diff
@@ -379,6 +381,7 @@
     "ResourceGroupName": "test-rg",
     "Type": "Microsoft.App/containerApps",
     "ResourceType": "Microsoft.App/containerApps",
+    "apiVersion": "2023-04-01-preview",
```
Comment (BernieWhite) [inline tests/PSRule.Rules.Azure.Tests/Resources.ContainerApp.json:384]:
```suggestion
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Thanks @BenjaminEngeset. Great work.

I have made the following updates:

- I renamed the rule to short it to 35 characters.
- Updated `RuleDataExportVisitor` to allow the `apiVersion` property to be exposed from IaC.

Let's remove `apiVersion` from one of the resource (i've provided a suggestion on this) so that we test for the case when no `apiVersion` is set, which should pass. This will be the case we need to handle for _in-flight_.

Double check the changes and let me know if you are fine with these.

**Comment by BenjaminEngeset** [comment]
@BernieWhite Looks great. I'm happy, all set!

Btw, the rule did work for me locally without the update in `RuleDataExportVisitor`, is that something that is used only for in-flight?

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. All good to merge.

**Comment by BernieWhite** [comment]
> @BernieWhite Looks great. I'm happy, all set!
> 
> Btw, the rule did work for me locally without the update in `RuleDataExportVisitor`, is that something that is used only for in-flight?

When PSRule expands resources for pre-flight from code this should apply.

**Pull request merged**